bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/handler/QbusDimmerHandler.java

```diff
@@ -0,0 +1,307 @@
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Error while initializing the thing.");
+        }
+    }
+
+    /**
+     * Returns the serial number of the controller
+     *
+     * @return the serial nr
+     */
+    public @Nullable String getSN() {
+        return this.sn;
+    }
+
+    /**
+     * Sets the serial number of the controller
+     */
+    public void setSN() {
+        QbusBridgeHandler QBridgeHandler = getBridgeHandler("Dimmer", dimmerId);
+        if (QBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No communication with Qbus Bridge!");
+            return;
+        }
+        this.sn = QBridgeHandler.getSn();
+        ;
+    }
+
+    /**
+     * Handle the status update from the thing
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        QbusCommunication QComm = getCommunication("Dimmer", dimmerId);
+
+        if (QComm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Bridge communication not initialized when trying to execute command for dimmer " + dimmerId);
+            return;
+        }
+
+        Map<Integer, QbusDimmer> dimmerComm = QComm.getDimmer();
+        if (dimmerComm != null) {
+            QbusDimmer QDimmer = dimmerComm.get(dimmerId);
+
+            if (QDimmer == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Bridge communication not initialized when trying to execute command for dimmer " + dimmerId);
+                return;
+            } else {
+
+                scheduler.submit(() -> {
+                    if (!QComm.communicationActive()) {
+                        restartCommunication(QComm, "Dimmer", dimmerId);
+                    }
+
+                    if (QComm.communicationActive()) {
+
+                        if (command == REFRESH) {
+                            handleStateUpdate(QDimmer);
+                            return;
+                        }
+
+                        switch (channelUID.getId()) {
+                            case CHANNEL_SWITCH:
+                                handleSwitchCommand(QDimmer, command);
+                                updateStatus(ThingStatus.ONLINE);
+                                break;
+
+                            case CHANNEL_BRIGHTNESS:
+                                handleBrightnessCommand(QDimmer, command);
+                                updateStatus(ThingStatus.ONLINE);
+                                break;
+
+                            default:
+                                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "Channel unknown " + channelUID.getId());
```

Comment:
It seems a bit unreasonable to set the Thing offline, when the user sent an invalid command to it. Same for the others.